lib/src/chatbox.dart

```diff
@@ -52,6 +53,30 @@ class MessageActionEvent {
     message = Message.fromJson(json['message']);
 }
 
+enum UrlNavigationAction { allow, deny }
+
+class UrlNavigationRequest {
+  final Uri? url;
+  final String rawUrl;
+  final bool isValidUrl;
+
+  UrlNavigationRequest._(
+    this.url, 
+    this.rawUrl,
+    this.isValidUrl,
+  ) ;
+
+  factory UrlNavigationRequest(String url){
```

Comment:
We should not do the String to Uri conversion in a public API.
A simple constructor with an Uri parameter is the best here.

The whole String to Uri conversion is better to be done in the private _shouldNavigateToUrl function.